src/App.tsx

```diff
@@ -1,14 +1,71 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Status } from './utils/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loaderIsShown, setLoaderIsShown] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const [todosStatus, setTodosStatus] = useState(Status.ALL);
+  const [query, setQuery] = useState('');
+
+  setTimeout(() => {
+    setLoaderIsShown(false);
+  }, 300);
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        let filteredTodos = todosFromServer;
+
+        switch (todosStatus) {
+          case Status.ACTIVE:
+            filteredTodos = filteredTodos.filter(todo => !todo.completed);
+            break;
+
+          case Status.COMPLETED:
+            filteredTodos = filteredTodos.filter(todo => todo.completed);
+            break;
+
+          default:
+            filteredTodos = [...filteredTodos];
+        }
+
+        const lowerQuery = query.toLowerCase();
+
+        filteredTodos = filteredTodos.filter(
+          todo => todo.title.toLowerCase().includes(lowerQuery),
```

Comment:
let's separate the logic, just fetch the todos here and filter them in another function
this will greatly improve readability